tests/phpunit/unit-tests/Helper/Fields/Test_Field_Repeater.php

```diff
@@ -0,0 +1,45 @@
+<?php
+declare( strict_types=1 );
+
+namespace GFPDF\Helper\Fields;
+
+use WP_UnitTestCase;
+use GFAPI;
+use GPDFAPI;
+
+/**
+ * @package     Gravity PDF
+ * @copyright   Copyright (c) 2022, Blue Liquid Designs
+ * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
+ */
+
+/**
+ * Class Test_Field_Repeater
+ *
+ *
+ * @group   helper
+ * @group   fields
+ */
+class Test_Field_Repeater extends WP_UnitTestCase {
+
+	/**
+	 * Test the Gravity PDF Repeater field add_entry_meta method
+	 */
+	public function test_repeater_add_entry_meta_method() {
+		$form  = $GLOBALS['GFPDF_Test']->form['repeater-consent-form'];
+		$entry = $GLOBALS['GFPDF_Test']->entries['repeater-consent-form'][0];
+
+		$form_id          = GFAPI::add_form( $form );
+		$entry['form_id'] = $form_id;
+		$entry_id         = GFAPI::add_entry( $entry );
+
+		$repeater = new Field_Repeater( new \GF_Field_Repeater( $form['fields'][1] ), GFAPI::get_entry( $entry_id ), GPDFAPI::get_form_class(), GPDFAPI::get_misc_class() );
+		$items = $repeater->add_entry_meta( [ 'field_sample_1' => 1, 'field_sample_2' => 2, 'created_by' => 'tester', 'source_url' => 'test.com' ] );
+
+		$this->assertEquals( 1, $items['field_sample_1']);
+		$this->assertEquals( 'tester', $items['created_by']);
+		$this->assertEquals( 'test.com', $items['source_url']);
+		// make sure that add_entry_meta returns an array with the expected order.
+		$this->assertEquals( 'field_sample_2', key(array_slice($items, -1, 1, true)) );
```

Comment:
Running `$items` through `array_values()` and assigning to a new variable, then doing multiple assertions to ensure the order is correctly would be clearer than this one-liner.